eth_crowdfund_be/eth_crowdfund_api/src/tests/test_campaign.py

```diff
@@ -0,0 +1,155 @@
+import pdb
+import unittest 
+import os 
+import json 
+# import flask_sqlalchemy
+from ..models.Campaign import Campaign
+from ..app import create_app, db
+
+def func(x):
+  return x + 1
+# basic test
+def test_answer():
+  assert func(3) == 4
+
+class CampaignTest(unittest.TestCase):
+  # set up the tests with 2 campaigns
+  # This runs before EVERY test. not once before all
+  def setUp(self):
+    self.app = create_app('testing')
+    self.client = self.app.test_client
+    self.campaign1 = {
+        "name": "camp",
+        "description": "desc1",
+        "image": "image.edu",
+        "contributors": 1,
+        "upvote": 2,
+        "manager": "3",
+        "address": "4",
+        "min_contribution": 5.0
+    }
+    self.campaign2 = {
+        "name": "Eth",
+        "description": "desc2",
+        "image": "image.com",
+        "contributors": 100,
+        "upvote": 222,
+        "manager": "13",
+        "address": "14",
+        "min_contribution": 50.2
+    }
+    
+    with self.app.app_context():
+      # create all db objects
+      db.create_all()
+      # delete test campaigns by name
+      existing_campaign1 = Campaign.get_campaign_by_name(self.campaign1["name"])
+      if existing_campaign1:
+        existing_campaign1.delete()
+      existing_campaign2 = Campaign.get_campaign_by_name(self.campaign2["name"])
+      if existing_campaign2:
+        existing_campaign2.delete()
+
+  def test_campaign_creation(self):
+    # send a request
+    res = self.client().post('/api/v1/campaigns/',
+                              headers={'Content-Type': 'application/json'},
+                              data=json.dumps(self.campaign1))
+    json_data = json.loads(res.data)
+    # check the names. I'd prefer to have a simpler call, but there are a few differences that prevent self.campaign1 == json_data
+    self.assertEqual(self.campaign1["name"], json_data["name"])
+    self.assertEqual(self.campaign1["description"], json_data["description"])
+    self.assertEqual(self.campaign1["image"], json_data["image"])
+    self.assertEqual(self.campaign1["contributors"], json_data["contributors"])
+    self.assertEqual(self.campaign1["upvote"], json_data["upvote"])
+    self.assertEqual(self.campaign1["manager"], json_data["manager"])
+    self.assertEqual(self.campaign1["address"], json_data["address"])
+    self.assertEqual(self.campaign1["min_contribution"], json_data["min_contribution"])
+    # just checking a sad path
+    self.assertNotEqual(self.campaign1["min_contribution"], 100)
+    self.assertEqual(res.status_code, 201)
+
+  def test_campaign_update(self):
+    campaign1_update = {
+      "description": "new description", 
+      "contributors": 5,
+      "manager": "1x3y"
+    }
+    # create the user before updating
```

Comment:
Is this leftover from a copy & paste? Is it referring to creating a campaign before updating?

Reply:
ahh, yes. I was looking at a tutorial for creating users. I'll modify it to say campaigns